src/main/java/org/broadinstitute/hellbender/engine/AbstractIntervalTreeBasedConcordanceWalker.java

```diff
@@ -0,0 +1,486 @@
+    public static final String CONFIDENCE_REGION_LONG_NAME = "confidence";
+
+    // The distance in bases to look ahead and cache when querying feature sources.
+    public static final int CACHE_LOOKAHEAD = 100_000;
+
+    public static final String FEATURE_NAME_FOR_TRUTH = "truth";
+
+    // ARGUMENTS =======================================================================================================
+
+    @Argument(doc = "TO BE IMPLEMENTED",
+            shortName = CONFIDENCE_REGION_SHORT_NAME,
+            fullName= CONFIDENCE_REGION_LONG_NAME,
+            optional = true)
+    protected String highConfidenceRegion;
+
+    @Argument(doc = "path to a file holding mask on region to be excluded",
+            shortName = "mask",
+            fullName = "ref-region-mask-file",
+            optional = true)
+    protected String refRegionMaskFilePath;
+
+    @Argument(doc = "A VCF containing truth variants",
+            shortName = TRUTH_VARIANTS_LONG_NAME,
+            fullName = TRUTH_VARIANTS_LONG_NAME)
+    public String truthVariantsFile;
+
+    @Argument(doc = "One VCF file containing call sets to be evaluated",
+            shortName = EVAL_VARIANTS_SHORT_NAME,
+            fullName = EVAL_VARIANTS_LONG_NAME)
+    public String evalVariantsFile;
+
+    // FIELDS ==========================================================================================================
+
+    protected SVIntervalTree<String> maskedOut;
+    protected SAMSequenceDictionary refSeqDict;
+    private VariantContextComparator variantContextComparator;
+
+    private FeatureDataSource<VariantContext> truthVariants;
+    protected SVIntervalTree<List<TruthVariant>> truthTree;
+
+    private FeatureDataSource<VariantContext> evalVariants;
+
+    /**
+     * For testing two objects of type T, using information stored in object of type R.
+     * See example in {@link #getConcordanceTester()}
+     */
+    @FunctionalInterface
+    public interface TriPredicate<T, R> {
+
+        Boolean test(T t1, T t2, R r);
+
+        default TriPredicate<T, R> and(final TriPredicate<? super T, ? super R> other) {
+            Objects.requireNonNull(other);
+            return (T t1, T t2, R r) -> test(t1, t2, r) && other.test(t1, t2, r);
+        }
+
+        default TriPredicate<T, R> negate() {
+            return (T t1, T t2, R r) -> !test(t1, t2, r);
+        }
+
+        default TriPredicate<T, R> or(final TriPredicate<? super T, ? super R> other) {
+            Objects.requireNonNull(other);
+            return (T t1, T t2, R r) -> test(t1, t2, r) || other.test(t1, t2, r);
+        }
+    }
+
+    protected TriPredicate<VariantContext, ReferenceContext> variantConcordanceTester;
+
+    // TRIVIA ==========================================================================================================
+
+    @Override
+    public final SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+        return refSeqDict;
```

Comment:
Is is a bit dangerous as discussed earlier as there is no guarateed that refSeqDict has been initialized at this point. Is a bit difficult since refSeqDict initialization requires truthVariants initialization. 

Perhaps you would need to call initializeTruthVariants() here and the in initializeTruthVariants() check whether it has been already initialized at the begining so you don't do it twice. 

This method could check whether refSeqDict is not null and if so just return its value.

Reply:
new implementation does the initialization correctly now, I think